web/models/campaign.ex

```diff
@@ -18,8 +23,9 @@ defmodule Jingle.Campaign do
   """
   def changeset(struct, params \\ %{}) do
     struct
-    |> cast(params, [:sponsor_id, :podcast_id, :start_date, :end_date, :copy, :zone])
-    |> validate_required([:sponsor_id, :podcast_id, :start_date, :end_date, :copy, :zone])
+    |> cast(params, [:sponsor_id, :podcast_id, :start_date, :end_date, :due_date, :original_copy, :edited_copy, :must_say, :notes, :approved, :zone])
+    |> validate_required([:start_date, :end_date, :original_copy, :zone])
```

Comment:
Sponsor/podcast relations no longer required?

Reply:
next lines take care of!:    
```elixir
    |> assoc_constraint(:sponsor)
    |> assoc_constraint(:podcast)
```

Reply:
ah, gotcha.  good call!